doc/api/stream.md

```diff
@@ -1532,6 +1532,43 @@ readable.on('data', (chunk) => {
 });
 ```
 
+### stream.Transform.by(asyncGeneratorFunction, [options])
+
+* `asyncGeneratorFunction` {AsyncGeneratorFunction} A mapping function which
+accepts a `source` iterable which can be used to read incoming data, while
+transformed data is pushed to the stream with the `yield` keyword.
+* `options` {Object} Options provided to `new stream.Transform([options])`.
+By default, `Transform.by()` will set `options.objectMode` to `true`,
+unless this is explicitly opted out by setting `options.objectMode` to `false`.
+
+A utility method for creating Transform Streams with async generator functions.
+The async generator is supplied a single argument, `source`, which is used to
+read incoming chunks.
+
+Yielded values become the data chunks emitted from the stream.
+
+```js
+const { Readable, Transform } = require('stream');
+
+const readable = Readable.from(['hello', 'streams']);
+async function * mapper(source) {
+  for await (const chunk of source) {
+    yield chunk.toUpperCase();
```

Comment:
I feel like it might make sense to mention that this relies on `Readable.from` and `Transform.by` working in object mode (and ditto for the `Readable.from` example, but that isn’t touched here)?

Reply:
I could put a comment in the code to reinforce? e.g. 
```
// if objectMode was set to false, the buffer would have to be converted to a string here
// but since it is true by default for both Readable.from and Transform.by each chunk is
// already a string 

```

Reply:
@addaleax does this satisfy?

Reply:
@davidmarkclements Yes, that text works for me :)